### DISCUSSION TITLE
ABL Mesh not Centered

### DISCUSSION THREAD

**Discussion by Larry2368** [synman/Ender-3-S1-Pro-Firmware#65] [Q&A]
After installing the 2.1.2-ender-3-s1 firmware update on my Ender 3 S1 Pro, I noticed the Auto Bed Leveling mesh is not centered on the bed.  I did check the xy offset and they are M851 Probe Offset X-31.5000 Y-41.8000 Z-3.0600 ; (mm).  I still learning this 3D printer stuff so any help would be appreciated.

Larry

**Comment by Braincake87** [comment]
A bit of a late reply, but I'm having the same issue, how did you solve it?

**Comment by Larry2368** [comment]
I have not solved it.  I have not had an opportunity to actually verify how
much the mesh is off.

I did use the bed visualization tool in OctoPrint to better square/tram my
bed.  So it has not been a huge issue.

**Reply by Braincake87** [reply]
Thanks for the quick reply!
I have the same situation, it’s a bit weird that the grid is off, but it works fine for printing.
Then I’ll just accept it as is…

**Comment by Larry2368** [comment]
I plan to figure it out.  Just keep getting interrupted by printing
things.  If I have another printer, it might be easier to get to.

I will try to remember to keep you posted.

**Reply by Braincake87** [reply]
Thanks.
In the meantime, I have updated my printer and screen to Thomas Toka's version, which doesn't have this mesh offset problem.
It can also do a lot more and has a [recent firmware release](https://github.com/ThomasToka/MarlinFirmware/releases) (4 days ago) and quite active support from Thomas himself through Discord.

**Comment by Larry2368** [comment]
What printer and version firmware do you have?

I have an Ender 3 S1 Pro running bugfix-2.1.x-F4-s1pro_ulb25.

**Reply by Braincake87** [reply]
I have an Ender 3 S1 Pro with an F1 board, had the 2.0.8 ABL version installed but now moved to Thomas Toka’s firmware, which doesn’t have this offset issue.

**Comment by Larry2368** [comment]
Sorry, I think that last reply from me got delayed.

I saw the Thomas Toka’s firmware comment a day or two ago.

Did the mesh appear to be positioned correctly on the bed even though
probing was significantly off?  I'm beginning to suspect probing and the
application of the mesh are both off as can be seen in the attached image.

What do you like and dislike about Thomas Toka’s firmware?  My printer time
is limited, so I have to prioritize accordingly.  But I have had several
print come out less perfect than I desire, so updating the firmware maybe a
good idea.

I fear with the explosion of newer, faster, and bigger printers the S1 Pro
is getting forgotten about.

Keep the filament flowing,
Larry

**Reply by Braincake87** [reply]
I am not sure how to assess the positioning of the ABL mesh. I just saw it as a visual in OctoPrint.

I like about TT firmware that it updates the display too and offers a lot more features there.
I use both OctoPrint and the display so for me that’s a huge plus.

**Comment by Larry2368** [comment]
If I read the install instructions correctly, I need to update to Cteality
2.0.8.28 before I can update to Thomas Toka’s newest release, even though I
running Marlin 2.1.x.bugfixes.  Is that the way you read it?

While this isn't overly complicated, I'm currently juggling several
projects and just don't want to spend time doing unnecessary steps.

Thanks,
Larry

**Reply by Braincake87** [reply]
I also read it that way, however I have an F1 board for which the 2.0.28 isn’t available.
I flashed the 2.0.24 from Creality and then the latest release from TT for the S1 Pro with F1 board. I used the ABL version.
It flashed fine, did the ABL fine and I ran several prints with it already.

**Comment by Larry2368** [comment]
Anything you don't like about TT's update.

Larry
